CMake/R/ConfigureRCPP.cmake

```diff
@@ -0,0 +1,93 @@
+# ConfigureRCPP.cmake: generate an mlpack .cpp file for a R binding given
+# input arguments.
+#
+# This file depends on the following variables being set:
+#
+#  * PROGRAM_NAME: name of the binding
+#  * PROGRAM_MAIN_FILE: the file containing the mlpackMain() function.
+#  * R_CPP_IN: path of the r_method.cpp.in file.
+#  * R_CPP_OUT: name of the output .cpp file.
+include("${SOURCE_DIR}/CMake/StripType.cmake")
+strip_type("${PROGRAM_MAIN_FILE}")
+
+# Extract the requried part from *main.cpp.
+# Example: mlpack/methods/adaboost/adaboost_main.cpp
+string(REGEX MATCHALL "\\/src\\/[A-Za-z\\/._]*" INCLUDE_MODEL_SAFE 
+    "${PROGRAM_MAIN_FILE}")
+string(REGEX REPLACE "\\/src\\/" "" INCLUDE_FILE "${INCLUDE_MODEL_SAFE}")
```

Comment:
Ahh, I accidentally broke this regex because my repository is in `/home/ryan/src/mlpack/`, so this results in files like `src/adaboost.cpp` having includes like this:

```
#include <mlpackmlpack/methods/adaboost/adaboost_main.cpp>
```

The suggestion is one way to fix it; you can also try some other way if you like.  This approach is at least a little safer since it strips off the name given exactly as it is in `SOURCE_DIR`.

```suggestion
# Extract the required part from *main.cpp.
# Example: mlpack/methods/adaboost/adaboost_main.cpp
string(REGEX REPLACE "${SOURCE_DIR}\\/src\\/" "" INCLUDE_FILE 
    "${PROGRAM_MAIN_FILE}")
```